app/javascript/mastodon/web_push_subscription.js

```diff
@@ -0,0 +1,137 @@
+
+          console.log('VAPID public key is unchanged:', subscriptionServerKey === currentServerKey);
+          console.log('Backend has same endpoint:', subscription.endpoint === serverEndpoint);
+
+          if (!serverEndpoint) {
+            console.log('Hm, the subscriptions seems to have removed in the backend...');
+          }
+
+          // If the VAPID public key did not change and the endpoint corresponds
+          // to the endpoint saved in the backend, the subscription is valid
+          if (subscriptionServerKey === currentServerKey && subscription.endpoint === serverEndpoint) {
+            return subscription;
+          } else {
+            if (subscription.endpoint !== serverEndpoint) {
+              console.log('Backend endpoint:', serverEndpoint);
+              console.log('Subscr. endpoint:', subscription.endpoint);
+            }
+
+            console.log('Unsubscribing and subscribing...');
+            // Something went wrong, try to subscribe again
+            return unsubscribe({ registration, subscription }).then(subscribe).then(sendSubscriptionToBackend);
+          }
+        }
+
+        console.log('Not subscribed, subscribing...');
+
+        // No subscription, try to subscribe
+        return subscribe(registration).then(sendSubscriptionToBackend);
+      })
+      .then(subscription => {
+        // If we got a PushSubscription (and not a subscription object from the backend)
+        // it means that the backend subscription is valid (and was set during hydration)
+        if (!(subscription instanceof PushSubscription)) {
+          console.log('Got a new subscription from the backend, dispatching action to save it in the store.');
+          store.dispatch(setSubscription(subscription));
+        } else {
+          console.log('Got a PushSubscription, which is already saved in the store.');
+        }
+      })
+      .catch(error => {
+        if (error.code === 20 && error.name === 'AbortError') {
+          // eslint-disable-next-line no-console
+          console.warn('Your browser supports Web Push Notifications, but does not seem to implement the VAPID protocol.');
+        } else if (error.code === 5 && error.name === 'InvalidCharacterError') {
+          // eslint-disable-next-line no-console
+          console.error('The VAPID public key seems to be invalid:', getApplicationServerKey());
+        } else {
+          console.error('Something went wrong:', error);
+        }
+
+        // Clear alerts and hide UI settings
+        console.log('Clearing subscription from the store...');
+        store.dispatch(clearSubscription());
+
+        try {
+          console.log('Fetching subscription and unsubscribing...');
+          getRegistration()
+            .then(getPushSubscription)
+            .then(unsubscribe);
+        } catch (e) {
+          console.error('Something went wrong yet again:', error);
+        }
+      });
+  } else {
+    // eslint-disable-next-line no-console
```

Comment:
I think you are aware of, but here I note you should be consistent, whether you use `eslint-disable-next-line` or not. If you use, apply for all other `console`. If you do not, remove this.